async_firebase/messages.py

```diff
@@ -215,3 +214,54 @@ class PushNotification:
 
     message: Message
     validate_only: t.Optional[bool] = field(default=False)
+
+
+class FcmPushResponse:
+    """The response received from an individual batched request to the FCM API.
+
+    The interface of this object is compatible with SendResponse object of
+    the Google's firebase-admin-python package.
+    """
+
+    def __init__(self, fcm_response: t.Optional[t.Dict[str, str]] = None, exception=None):
+        """Inits FcmPushResponse object.
+
+        :param fcm_response: a dictionary with the data that FCM returns as a payload
+        :param exception: an exception that may happen when communicating with FCM
+        """
+        self.message_id = fcm_response.get("name") if fcm_response else None
+        self.exception = exception
+
+    @property
+    def success(self) -> bool:
+        """A boolean indicating if the request was successful."""
+        return self.message_id is not None and not self.exception
```

Comment:
This double-check seems redundant you may leave only on condition.

Reply:
Actually, the object may be constructed incorrectly, and in this case `success` attribute value will indicate this.